### ISSUE TITLE
Crashing tests with gcc 5.2.1

### ISSUE THREAD

**Issue by springmeyer** [mapnik/mapnik#3103]
Reported by @barjac and @spuhler originally at #3072. Likely related or the same as #3090.

Relevant logs are https://gist.github.com/springmeyer/4d407ce524feafbb1301

**Comment by talaj** [comment]
I think it's all related to non-release distributions of Debian. I'm trying to localize the problem.

**Comment by talaj** [comment]
Hm, but looking into you log and it seems like some RPM distro...

**Comment by talaj** [comment]
https://github.com/SFML/SFML/issues/936 seems to be the same issue.

**Comment by spuhler** [comment]
We finally got the 64 bit package to build with pointer errors in the test. See log
http://mtf.no-ip.co.uk/pub/linux/barjac/distrib/cauldron/x86_64/log/mapnik-3.0.5-1.mga6.src.rpm/build.0.20151006222831.log
32bit doesn't build because of running out of memory. Has anybody been able to build it?
cc1plus: out of memory allocating 16777204 bytes after a total of 172331008 bytes
look at the end of the log,
http://pkgsubmit.mageia.org/uploads/failure/cauldron/core/release/20151006202534.spuhler.valstar.6668/log/mapnik-3.0.5-1.mga6/build.0.20151006202602.log

**Comment by kennykb** [comment]
Still crashing in 3.0.8:

<pre>
$ make test
* Starting Mapnik tests
* Running C++ Unit tests...

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
run is a Catch v1.1 b1 host application.
Run with -? for options

-------------------------------------------------------------------------------
parameters
  get/set
-------------------------------------------------------------------------------
test/unit/core/params_test.cpp:28
...............................................................................

test/unit/core/params_test.cpp:41: FAILED:
  REQUIRE( (params.get<mapnik::boolean_type>("bool") && *params.get<mapnik::boolean_type>("bool") == true) )
with expansion:
  false

*** Error in `./test/unit/run': munmap_chunk(): invalid pointer: 0x00007fffec4a9a58 ***
test/unit/core/params_test.cpp:30: FAILED:
  {Unknown expression after the reported line}
due to a fatal error condition:
  SIGABRT - Abort (abnormal termination) signal

===============================================================================
test cases:    22 |    21 passed | 1 failed
assertions: 17411 | 17409 passed | 2 failed

* Running standalone C++ tests...
===============================================================================
All tests passed (2 assertions in 1 test case)

===============================================================================
All tests passed (4 assertions in 1 test case)

===============================================================================
All tests passed (61 assertions in 1 test case)

* Running visual tests...
*** Error in `./test/visual/run': free(): invalid pointer: 0x00007ffe424bfff8 ***
./test/run: line 42:  5677 Aborted                 (core dumped) ./test/visual/run -j $JOBS
Makefile:89: recipe for target 'test' failed
make: *** [test] Error 128
</pre>

**Comment by artemp** [comment]
@kennykb - this has been fixed after  3.0.8 (2d15567)

**Comment by springmeyer** [comment]
yes, solved by #3146
